sdk/tables/azure-data-tables/tests/test_table_batch_cosmos.py

```diff
@@ -474,6 +478,56 @@ def test_delete_batch_with_bad_kwarg(self, tables_cosmos_account_name, tables_pr
         finally:
             self._tear_down()
 
+    @pytest.mark.skipif(sys.version_info < (3, 0), reason="requires Python3")
+    @pytest.mark.live_test_only
+    @cosmos_decorator
+    def test_batch_sas_auth(self, tables_cosmos_account_name, tables_primary_cosmos_account_key):
+        # Arrange
+        self._set_up(tables_cosmos_account_name, tables_primary_cosmos_account_key, url="cosmos")
+        try:
+            token = self.generate_sas(
+                generate_table_sas,
+                tables_primary_cosmos_account_key,
+                self.table_name,
+                permission=TableSasPermissions(add=True, read=True, update=True, delete=True),
+                expiry=datetime.utcnow() + timedelta(hours=1),
+                start=datetime.utcnow() - timedelta(minutes=1),
+            )
+            token = AzureSasCredential(token)
+
+            # Act
+            service = TableServiceClient(
+                self.account_url(tables_cosmos_account_name, "cosmos"),
+                credential=token,
+            )
+            table = service.get_table_client(self.table_name)
+
+            entity = TableEntity()
+            entity['PartitionKey'] = 'batch_inserts'
+            entity['test'] = EntityProperty(True, EdmType.BOOLEAN)
+            entity['test2'] = 'value'
+            entity['test3'] = 3
+            entity['test4'] = EntityProperty(1234567890, EdmType.INT32)
+# 'https://seankaneprim.table.core.windows.net/$batch?st=2021-05-18T16%3A19%3A58Z&se=2021-05-18T17%3A20%3A58Z&sp=raud&sv=2019-02-02&tn=uttabled4f0e8f&sig=cROztTVf0vKXUE1IQyyJpnjA3Ua1a4NMYYF0VnCMt/4%3D'
+# 'https://seankaneprim.table.core.windows.net/$batch?st=2021-05-18T16%3A20%3A57Z&se=2021-05-18T17%3A21%3A57Z&sp=raud&sv=2019-02-02&tn=uttable81841182&sig=Czf9WOvwZT790ZaUqOPanvpa04mtBMQ28xvAr/l%2BPbM%3D'
```

Comment:
We should avoid putting the sas tokens in the tests as well ;)

Reply:
Ahhh that was dumb, I was comparing when i had some failures and never removed them. Big mistake on my part